File: Kattis/ferskastajarmid/ferskastajarmid.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

int main() {
    int n;
    std::cin >> n;
    std::vector<std::pair<std::string, int>> memes;
    memes.reserve(n);

    int bestScore = 0;

    while (n > 0) {
        std::string meme;
        int contro, cool, fresh;
        std::cin >> meme >> contro >> cool;
        fresh = contro * cool;
        if (bestScore <= fresh) {
            bestScore = fresh;
            memes.push_back({meme, fresh});
        }
        n--;
    }

    std::sort(memes.begin(), memes.end(), [](const auto& a, const auto& b) {
        if (a.second == b.second)
            return a.first < b.first;
        return a.second > b.second;
    });

    std::cout << memes[0].first << "\n";
    return 0;
}

File: Kattis/ferskastajarmid/ferskastajarmid.py
n = int(input())
memes = []
bestScore = 0

while n > 0:
    meme, contro, cool = input().split()
    contro = int(contro)
    cool = int(cool)
    fresh = contro * cool

    if bestScore <= fresh:
        bestScore = fresh
        memes.append((meme, fresh))

    n -= 1

memes.sort(key=lambda x: (-x[1], x[0]))

print(memes[0][0])